prime-router/src/main/kotlin/cli/tests/BasicTests.kt

```diff
@@ -115,125 +138,162 @@ class End2EndUniversalPipeline : CoolTest() {
                 payloadName = "$name ${status.description}",
             )
         if (responseCode != HttpURLConnection.HTTP_CREATED) {
-            bad("***end2end_up Test FAILED***:  response code $responseCode")
-            passed = false
+            bad("***$name FAILED***: Response code $responseCode")
         } else {
-            good("Posting of async report succeeded with response code $responseCode")
+            good("Posting of report succeeded with response code $responseCode")
         }
 
         echo(json)
-        passed = passed and examinePostResponse(json, false)
-        if (!passed) {
-            bad("***async end2end_up FAILED***: Error in post response")
+
+        if (!examinePostResponse(json, false)) {
+            bad("***$name FAILED***: Error in post response")
         }
-        // gets back 'received' reportId
-        val reportId = getReportIdFromResponse(json)
 
-        if (reportId != null) {
-            // check convert step
-            val convertResults = pollForStepResult(reportId, TaskAction.convert)
-            // verify each result is valid
-            for (result in convertResults.values)
-                passed = passed && examineStepResponse(result, "convert", sender.topic)
-            if (!passed) {
-                bad("***async end2end_up FAILED***: Convert result invalid")
+        return getReportIdFromResponse(json)
+    }
+
+    private suspend fun end2EndUP(
+        file: File,
+        environment: Environment,
+        sender: Sender,
+        expectedReceivers: List<Receiver>,
+        reportId: ReportId,
+    ): Boolean {
+        var passed = true
+        val connectionInfo = "DefaultEndpointsProtocol=http;AccountName=devstoreaccount1;" +
+            "AccountKey=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==;" +
+            "BlobEndpoint=http://localhost:10000/devstoreaccount1;" +
+            "QueueEndpoint=http://localhost:10001/devstoreaccount1;"
+        val blobContainerMetadata = BlobAccess.BlobContainerMetadata("reports", connectionInfo)
+
+        // Call history endpoint to get results
+        val getUrl = "${environment.url}/api/waters/report/$reportId/history"
+        val (_: Int, response: String) = HttpUtilities.getHttp(getUrl)
+
+        expectedReceivers.forEach { expectedReceiver ->
+            // Retrieve external filenames from the history endpoint
+            val actualFilename = findReportExternalNames(response, expectedReceiver)
+            if (actualFilename.isNotEmpty()) {
+                good("The report was received by ${expectedReceiver.name}")
+            } else {
+                bad("***$name FAILED***: Did not find the sent report name for receiver ${expectedReceiver.name}")
+                passed = false
+                return@forEach
```

Comment:
Still wrapping my brain around Kotlin so let me know if there's a better way to do this. My thought here is that this part of the test happens in a few seconds, so the test should still go through the motion of checking each sent report even if one fails.